core/include/scipp/core/element/comparison.h

```diff
@@ -34,12 +43,17 @@ constexpr auto isclose_units = [](const units::Unit &x, const units::Unit &y,
 constexpr auto isclose = overloaded{
     transform_flags::expect_no_variance_arg_t<2>{}, isclose_types_t{},
     isclose_units, [](const auto &x, const auto &y, const auto &t) {
-      using std::abs;
-      return abs(x - y) <= t;
+      if constexpr (std::is_same_v<std::decay_t<decltype(x)>,
+                                   Eigen::Vector3d>) {
+        return (x - y).norm() <= t;
+      } else {
+        using std::abs;
+        return abs(x - y) <= t;
+      }
```

Comment:
Unfortunately ugly (and same `abs`/`norm` below in `variable/comparison.cpp`). Should we consider a function with a common name, e.g., overloaded `core::norm`, and `variable::norm` to work on numbers as well as vectors?

Reply:
One problem with pushing this down into `core::norm` is that you still will need the compile-time type check. In addition it becomes problematic (at least I can't find a good solution) to do the type comparisons, since `x - y` is an intermediate Eigen Binary operation type, not an `Eigen::MatrixBase` type. `decltype(x.eval())` type solutions quickly get very messy.